proxy-node-shared/src/test/java/uk/gov/ida/notification/shared/proxy/VSPProxyTranslateResponseTest.java

```diff
@@ -0,0 +1,149 @@
+import uk.gov.ida.notification.contracts.verifyserviceprovider.TranslatedHubResponseBuilder;
+import uk.gov.ida.notification.contracts.verifyserviceprovider.VerifyServiceProviderTranslationRequest;
+import uk.gov.ida.notification.contracts.verifyserviceprovider.VspLevelOfAssurance;
+import uk.gov.ida.notification.contracts.verifyserviceprovider.VspScenario;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriBuilder;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import static org.mockito.ArgumentMatchers.eq;
+
+@RunWith(MockitoJUnitRunner.class)
+public class VSPProxyTranslateResponseTest {
+
+    @Path("/")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public static class TestVSPResource {
+
+        @POST
+        @Path("/translate-response")
+        public TranslatedHubResponse testGetTranslatedHubResponse(VerifyServiceProviderTranslationRequest request) {
+            if ("SAMLResponse1234".equals(request.getSamlResponse())) {
+                return TranslatedHubResponseBuilder.getTranslatedHubResponseIdentityVerified();
+            } else {
+                return TranslatedHubResponseBuilder.getTranslatedHubResponseAuthenticationFailed();
+            }
+        }
+    }
+
+    @Path("/translate-response")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public static class TestVSPServerErrorResource {
+
+        @POST
+        public Response testGenerate(AuthnRequestGenerationBody authnRequestGenerationBody) {
+            return Response.serverError().build();
+        }
+    }
+
+    @Path("/translate-response")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public static class TestVSPClientErrorResource {
+
+        @POST
+        public Response testGenerate(AuthnRequestGenerationBody authnRequestGenerationBody) {
+            return Response.status(Response.Status.BAD_REQUEST).build();
+        }
+    }
+
+    @ClassRule
+    public static final DropwizardClientRule testVspClientRule = new DropwizardClientRule(new TestVSPResource());
+
+    @ClassRule
+    public static final DropwizardClientRule testVspServerErrorClientRule = new DropwizardClientRule(new TestVSPServerErrorResource());
+
+    @ClassRule
+    public static final DropwizardClientRule testVspClientErrorClientRule = new DropwizardClientRule(new TestVSPClientErrorResource());
+
+    @Spy
+    JsonClient jsonClient = new JsonClient(
+            new ErrorHandlingClient(ClientBuilder.newClient()),
+            new JsonResponseProcessor(new ObjectMapper())
+    );
+
+    @Test
+    public void shouldReturnTranslatedHubResponseAuthnFailed() {
```

Comment:
There are tests to check for the `IDENTITY_VERIFIED` and `AUTHENTICATION_FAILED`
What about tests for `CANCELLATION` and `REQUEST_ERROR`?

Reply:
Could do, but the behaviour is not actually any different for the other two statuses so the test wouldn't be doing much.

Reply:
OK you're right that's not needed for the VSP Proxy test, however, it should be check in a test for either `HubResponseTranslatorResource` or `HubResponseTranslator`